template/addons/azure/webapp_routing/ingress.yaml

```diff
@@ -2,32 +2,32 @@ apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   annotations:
-    kubernetes.azure.com/tls-cert-keyvault-uri: "{{ingress-tls-cert-keyvault-uri}}"
-    kubernetes.azure.com/use-osm-mtls: "{{ingress-use-osm-mtls}}"
+    kubernetes.azure.com/tls-cert-keyvault-uri: "{{index . "ingress-tls-cert-keyvault-uri"}}"
+    kubernetes.azure.com/use-osm-mtls: "{{index . "ingress-use-osm-mtls"}}"
     nginx.ingress.kubernetes.io/backend-protocol: HTTPS
     nginx.ingress.kubernetes.io/configuration-snippet: |2-
 
-      proxy_ssl_name "default.{{service-namespace}}.cluster.local";
+      proxy_ssl_name "default.{{index . "service-namespace"}}.cluster.local";
     nginx.ingress.kubernetes.io/proxy-ssl-secret: kube-system/osm-ingress-client-cert
```

Comment:
not super necessary right now since it shouldn't make a difference to the end functionality, but we shouldn't have this annotation/attempt to utilize it unless OSM is enabled.

Granted it seems like we're not recommending this use case so perhaps we don't need to make it an option for users to specify whether or not we have OSM.

Has this been manually tested?